### PR TITLE
KMP algo in python3

### PR DESCRIPTION
**Pull request by DaRkSouL-1**
**Hi!**  
Thanks for considering for contributing to this ever-growing list of algorithm and data structure implementations.
Your contribution is valuable. Use the follow template for creating PRs. (Delete this part)
---

**Fixes #{ISSUE_NUMBER}**

By submitting this pull request I confirm I've read and complied with the below declarations.

-   [ ] I have read the [Contribution guidelines](https://github.com/FOSS-UCSC/FOSSALGO/blob/master/CONTRIBUTING.md) 
and I am confident that my PR reflects them.

-   [ ] I have followed the [coding guidelines](https://github.com/FOSS-UCSC/FOSSALGO/blob/master/CONTRIBUTING.md#cs) 
for this project.

-   [ ] My code follows the [skeleton code structure](https://github.com/FOSS-UCSC/FOSSALGO/blob/master/CONTRIBUTING.md#sample).

-   [ ] This pull request has a descriptive title. 
For example, `Added {Algorithm/DS name} [{Language}]`, not `Update README.md` or `Added new code`.

-   [ ] This pull request shall only be merged once the build passes.

-   [ ] I have mentioned the issue number correctly (with hyper-link) in this pull request description.

### CODE REVIEW

File: algorithms/ar-kmp/python3/kmp algorithm python.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
+def KMPSearch(pat, txt):
```
Comment (RavinduSachintha) [inline algorithms/ar-kmp/python3/kmp algorithm python.py:1]: Codacy found an issue: [Trailing whitespace](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6361667)

File: algorithms/ar-kmp/python3/kmp algorithm python.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
+def KMPSearch(pat, txt): 
+	M = len(pat) 
+	N = len(txt) 
+	lps = [0]*M 
+	j = 0  
+	computeLPSArray(pat, M, lps) 
+
+	i = 0
```
Comment (RavinduSachintha) [inline algorithms/ar-kmp/python3/kmp algorithm python.py:8]: Codacy found an issue: [Trailing whitespace](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6361667)

File: algorithms/ar-kmp/python3/kmp algorithm python.py [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+	i = 0 
+	while i < N: 
+		if pat[j] == txt[i]: 
+			i += 1
+			j += 1
+
+		if j == M: 
+			print ("Found pattern at index " + str(i-j)) 
+			j = lps[j-1]
```
Comment (RavinduSachintha) [inline algorithms/ar-kmp/python3/kmp algorithm python.py:16]: Codacy found an issue: [Trailing whitespace](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6361667)

File: algorithms/ar-kmp/python3/kmp algorithm python.py [hunk 4]
```diff
@@ -0,0 +1,41 @@
+def KMPSearch(pat, txt): 
+	M = len(pat) 
+	N = len(txt) 
+	lps = [0]*M
```
Comment (RavinduSachintha) [inline algorithms/ar-kmp/python3/kmp algorithm python.py:4]: Codacy found an issue: [Trailing whitespace](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6361667)

File: algorithms/ar-kmp/python3/kmp algorithm python.py [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+	i = 0 
+	while i < N: 
+		if pat[j] == txt[i]: 
+			i += 1
+			j += 1
+
+		if j == M: 
+			print ("Found pattern at index " + str(i-j)) 
+			j = lps[j-1] 
+		elif i < N and pat[j] != txt[i]:
```
Comment (RavinduSachintha) [inline algorithms/ar-kmp/python3/kmp algorithm python.py:17]: Codacy found an issue: [Trailing whitespace](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6361667)

File: algorithms/ar-kmp/python3/kmp algorithm python.py [hunk 6]
```diff
@@ -0,0 +1,41 @@
+def KMPSearch(pat, txt): 
+	M = len(pat) 
+	N = len(txt) 
+	lps = [0]*M 
+	j = 0  
+	computeLPSArray(pat, M, lps) 
+
+	i = 0 
+	while i < N: 
+		if pat[j] == txt[i]:
```
Comment (RavinduSachintha) [inline algorithms/ar-kmp/python3/kmp algorithm python.py:10]: Codacy found an issue: [Trailing whitespace](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6361667)

File: algorithms/ar-kmp/python3/kmp algorithm python.py [hunk 7]
```diff
@@ -0,0 +1,41 @@
+def KMPSearch(pat, txt): 
+	M = len(pat) 
+	N = len(txt) 
+	lps = [0]*M 
+	j = 0  
+	computeLPSArray(pat, M, lps) 
+
+	i = 0 
+	while i < N:
```
Comment (RavinduSachintha) [inline algorithms/ar-kmp/python3/kmp algorithm python.py:9]: Codacy found an issue: [Trailing whitespace](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6361667)

File: algorithms/ar-kmp/python3/kmp algorithm python.py [hunk 8]
```diff
@@ -0,0 +1,41 @@
+def KMPSearch(pat, txt): 
+	M = len(pat)
```
Comment (RavinduSachintha) [inline algorithms/ar-kmp/python3/kmp algorithm python.py:2]: Codacy found an issue: [Trailing whitespace](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6361667)

File: algorithms/ar-kmp/python3/kmp algorithm python.py [hunk 9]
```diff
@@ -0,0 +1,41 @@
...
+			if j != 0: 
+				j = lps[j-1] 
+			else: 
+				i += 1
+
+def computeLPSArray(pat, M, lps): 
+	len = 0
+	lps[0]
+	i = 1
+	while i < M:
```
Comment (RavinduSachintha) [inline algorithms/ar-kmp/python3/kmp algorithm python.py:27]: Codacy found an issue: [Trailing whitespace](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6361667)

File: algorithms/ar-kmp/python3/kmp algorithm python.py [hunk 10]
```diff
@@ -0,0 +1,41 @@
...
+	lps[0]
+	i = 1
+	while i < M: 
+		if pat[i]== pat[len]: 
+			len += 1
+			lps[i] = len
+			i += 1
+		else: 
+			if len != 0: 
+				len = lps[len-1]
```
Comment (RavinduSachintha) [inline algorithms/ar-kmp/python3/kmp algorithm python.py:34]: Codacy found an issue: [Trailing whitespace](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6361667)

File: algorithms/ar-kmp/python3/kmp algorithm python.py [hunk 11]
```diff
@@ -0,0 +1,41 @@
...
+				j = lps[j-1] 
+			else: 
+				i += 1
+
+def computeLPSArray(pat, M, lps): 
+	len = 0
+	lps[0]
+	i = 1
+	while i < M: 
+		if pat[i]== pat[len]:
```
Comment (RavinduSachintha) [inline algorithms/ar-kmp/python3/kmp algorithm python.py:28]: Codacy found an issue: [Trailing whitespace](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6361667)

File: algorithms/ar-kmp/python3/kmp algorithm python.py [hunk 12]
```diff
@@ -0,0 +1,41 @@
...
+	computeLPSArray(pat, M, lps) 
+
+	i = 0 
+	while i < N: 
+		if pat[j] == txt[i]: 
+			i += 1
+			j += 1
+
+		if j == M: 
+			print ("Found pattern at index " + str(i-j))
```
Comment (RavinduSachintha) [inline algorithms/ar-kmp/python3/kmp algorithm python.py:15]: Codacy found an issue: [Trailing whitespace](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6361667)

File: algorithms/ar-kmp/python3/kmp algorithm python.py [hunk 13]
```diff
@@ -0,0 +1,41 @@
...
+		if j == M: 
+			print ("Found pattern at index " + str(i-j)) 
+			j = lps[j-1] 
+		elif i < N and pat[j] != txt[i]: 
+			if j != 0: 
+				j = lps[j-1] 
+			else: 
+				i += 1
+
+def computeLPSArray(pat, M, lps):
```
Comment (RavinduSachintha) [inline algorithms/ar-kmp/python3/kmp algorithm python.py:23]: Codacy found an issue: [Trailing whitespace](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6361667)

File: algorithms/ar-kmp/python3/kmp algorithm python.py [hunk 14]
```diff
@@ -0,0 +1,41 @@
...
+			print ("Found pattern at index " + str(i-j)) 
+			j = lps[j-1] 
+		elif i < N and pat[j] != txt[i]: 
+			if j != 0: 
+				j = lps[j-1] 
+			else: 
+				i += 1
+
+def computeLPSArray(pat, M, lps): 
+	len = 0
```
Comment (RavinduSachintha) [inline algorithms/ar-kmp/python3/kmp algorithm python.py:24]: Codacy found an issue: [Redefining built-in 'len'](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6361667)

File: algorithms/ar-kmp/python3/kmp algorithm python.py [hunk 15]
```diff
@@ -0,0 +1,41 @@
...
+	j = 0  
+	computeLPSArray(pat, M, lps) 
+
+	i = 0 
+	while i < N: 
+		if pat[j] == txt[i]: 
+			i += 1
+			j += 1
+
+		if j == M:
```
Comment (RavinduSachintha) [inline algorithms/ar-kmp/python3/kmp algorithm python.py:14]: Codacy found an issue: [Trailing whitespace](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6361667)

File: algorithms/ar-kmp/python3/kmp algorithm python.py [hunk 16]
```diff
@@ -0,0 +1,41 @@
+def KMPSearch(pat, txt): 
+	M = len(pat) 
+	N = len(txt) 
+	lps = [0]*M 
+	j = 0
```
Comment (RavinduSachintha) [inline algorithms/ar-kmp/python3/kmp algorithm python.py:5]: Codacy found an issue: [Trailing whitespace](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6361667)

File: algorithms/ar-kmp/python3/kmp algorithm python.py [hunk 17]
```diff
@@ -0,0 +1,41 @@
+def KMPSearch(pat, txt): 
+	M = len(pat) 
+	N = len(txt)
```
Comment (RavinduSachintha) [inline algorithms/ar-kmp/python3/kmp algorithm python.py:3]: Codacy found an issue: [Trailing whitespace](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6361667)

File: algorithms/ar-kmp/python3/kmp algorithm python.py [hunk 18]
```diff
@@ -0,0 +1,41 @@
...
+	len = 0
+	lps[0]
+	i = 1
+	while i < M: 
+		if pat[i]== pat[len]: 
+			len += 1
+			lps[i] = len
+			i += 1
+		else: 
+			if len != 0:
```
Comment (RavinduSachintha) [inline algorithms/ar-kmp/python3/kmp algorithm python.py:33]: Codacy found an issue: [Trailing whitespace](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6361667)

File: algorithms/ar-kmp/python3/kmp algorithm python.py [hunk 19]
```diff
@@ -0,0 +1,41 @@
...
+			j = lps[j-1] 
+		elif i < N and pat[j] != txt[i]: 
+			if j != 0: 
+				j = lps[j-1] 
+			else: 
+				i += 1
+
+def computeLPSArray(pat, M, lps): 
+	len = 0
+	lps[0]
```
Comment (RavinduSachintha) [inline algorithms/ar-kmp/python3/kmp algorithm python.py:25]: Codacy found an issue: [Statement seems to have no effect](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6361667)

File: algorithms/ar-kmp/python3/kmp algorithm python.py [hunk 20]
```diff
@@ -0,0 +1,41 @@
...
+			i += 1
+			j += 1
+
+		if j == M: 
+			print ("Found pattern at index " + str(i-j)) 
+			j = lps[j-1] 
+		elif i < N and pat[j] != txt[i]: 
+			if j != 0: 
+				j = lps[j-1] 
+			else:
```
Comment (RavinduSachintha) [inline algorithms/ar-kmp/python3/kmp algorithm python.py:20]: Codacy found an issue: [Trailing whitespace](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6361667)

**Comment by RavinduSachintha** [comment]
@DaRkSouL-1 the file name is not valid. please read the contribution guidelines and do the necessary changes. you need to resolve all the issues mentioned by the review system.

**Comment by Danushka96** [comment]
Closed Due to inactivity

**Pull request closed without merging**